terminal/display_windows.go

```diff
@@ -29,3 +29,22 @@ func EraseLine(out FileWriter, mode EraseLineMode) error {
 	_, _, err := procFillConsoleOutputCharacter.Call(uintptr(handle), uintptr(' '), uintptr(x), uintptr(*(*int32)(unsafe.Pointer(&cursor))), uintptr(unsafe.Pointer(&w)))
 	return normalizeError(err)
 }
+
+func EraseScreen(out FileWriter, mode EraseScreenMode) error {
+	handle := syscall.Handle(out.Fd())
+
+	var csbi consoleScreenBufferInfo
+	if _, _, err := procGetConsoleScreenBufferInfo.Call(uintptr(handle), uintptr(unsafe.Pointer(&csbi))); normalizeError(err) != nil {
+		return err
+	}
+
+	var w uint32
+	cursor := csbi.cursorPosition
+
+	lineCount := csbi.window.bottom - csbi.cursorPosition.Y
+	termWidth := csbi.size.X
+	screenSize := lineCount * termWidth
```

Comment:
The Windows version of this `EraseScreen` function clears the lines from the cursor (at [the beginning of the prompt](https://github.com/e-zim/survey/blob/d29d6fa9a7a31b354b33c29b6128b504fd3c6af6/renderer.go#L135)) to the end of the displayed window (which is the end of the outputted prompt). For reference, [the `consoleScreenBufferInfo` structure](https://learn.microsoft.com/en-us/windows/console/console-screen-buffer-info-str) contains terminal dimensions and coordinates of the window and cursor.

This seems to match the behavior of the POSIX `EraseScreen` for resetting the prompt, but doesn't account for the `EraseScreenMode`. Not sure if supporting the other modes (clearing to the beginning of the screen and the entire screen) is necessary for how this function is being used, but I'd be in favor of forgoing them to keep this implementation more simple. Definitely open to thoughts here!